airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -953,6 +954,11 @@ def create_concurrent_cursor_from_datetime_based_cursor(
             if "stream_state" not in kwargs
             else kwargs["stream_state"]
         )
+        if stream_state_migrations:
```

Comment:
nit: can we extract this to avoid duplication in `model_to_component_factory.py`?